src/CrossMappings/ccm.jl

```diff
@@ -113,21 +112,19 @@ function crossmap(x, y, d, τ, bootstrap_method::Symbol;
 
             # Depending on the input data, we might encounter a case where the distance from xᵢ
             # to its closest neighbor is indistinguishable from zero. In that case, we encounter
-            # division by zero when computing weights. In that case, we set all weights to zero.
+            # division by zero when computing weights. In that case, set weights to zero (but in
+            # practice, just continue to the next step, because ỹᵢ will is already initialized to 0)
             dist₁ = norm(xᵢ - nns[1])
+            dist₁ > 0.0 || continue
 
-            if dist₁ == 0.0
-                w .= 0.0
-            else 
-                for j = 1:d+1
-                    distⱼ = norm(xᵢ - nns[j])
-                    u[j] = exp(-distⱼ / dist₁)
-                end
-                for j = 1:d+1
-                    w[j] = u[j] / sum(u)
-                end
+            for j = 1:d+1
+                distⱼ = norm(xᵢ - nns[j])
+                u[j] = exp(-distⱼ / dist₁)
             end
-
+            for j = 1:d+1
+                w[j] = u[j] / sum(u)
+            end
```

Comment:
Why are you doing `sum(u)` here `d` times instead of only once? Seems to me that `sum(u)` doesn't depend on `j` but it is computed for every `j` nevertheless.

Reply:
Good point!